src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs

```diff
@@ -1,751 +1,5624 @@
+            stridesOrderRentedBuffer.Dispose();
+
+            return output;
+        }
+        #endregion
+
+        #region TryBroadcastTo
+        /// <summary>
+        /// Broadcast the data from <paramref name="tensor"/> to the smallest broadcastable shape compatible with <paramref name="destination"/> and stores it in <paramref name="destination"/>
+        /// If the shapes are not compatible, false is returned.
+        /// </summary>
+        /// <param name="tensor">Input <see cref="Tensor{T}"/>.</param>
+        /// <param name="destination">Destination <see cref="TensorSpan{T}"/>.</param>
+        public static bool TryBroadcastTo<T>(this Tensor<T> tensor, in TensorSpan<T> destination)
+        {
+            return tensor.AsReadOnlyTensorSpan().TryBroadcastTo(destination);
+        }
+
+        /// <summary>
+        /// Broadcast the data from <paramref name="tensor"/> to the smallest broadcastable shape compatible with <paramref name="destination"/> and stores it in <paramref name="destination"/>
+        /// If the shapes are not compatible, false is returned.
+        /// </summary>
+        /// <param name="tensor">Input <see cref="TensorSpan{T}"/>.</param>
+        /// <param name="destination">Destination <see cref="TensorSpan{T}"/>.</param>
+        public static bool TryBroadcastTo<T>(in this TensorSpan<T> tensor, in TensorSpan<T> destination)
+        {
+            return tensor.AsReadOnlyTensorSpan().TryBroadcastTo(destination);
+        }
+
+        /// <summary>
+        /// Broadcast the data from <paramref name="tensor"/> to the smallest broadcastable shape compatible with <paramref name="destination"/> and stores it in <paramref name="destination"/>
+        /// If the shapes are not compatible, false is returned.
+        /// </summary>
+        /// <param name="tensor">Input <see cref="ReadOnlyTensorSpan{T}"/>.</param>
+        /// <param name="destination">Destination <see cref="TensorSpan{T}"/>.</param>
+        public static bool TryBroadcastTo<T>(in this ReadOnlyTensorSpan<T> tensor, in TensorSpan<T> destination)
+        {
+            TensorOperation.ValidateCompatibility(tensor, destination);
+            if (!TensorShape.AreCompatible(destination._shape, tensor._shape, false))
+                return false;
+
+            BroadcastTo(tensor, destination);
+            return true;
+        }
+        #endregion
+
+        #region Unsqueeze
+        /// <summary>
+        /// Insert a new dimension of length 1 that will appear at the dimension position.
+        /// </summary>
+        /// <param name="tensor">The <see cref="Tensor{T}"/> to add a dimension of length 1.</param>
+        /// <param name="dimension">The index of the dimension to add.</param>
+        public static Tensor<T> Unsqueeze<T>(this Tensor<T> tensor, int dimension)
+        {
+            if (dimension > tensor.Lengths.Length)
+                ThrowHelper.ThrowArgument_AxisLargerThanRank();
+            if (dimension < 0)
```

Comment:
I'll not comment more on checking dimension input. Just in case if you consider doing anything, will be good to scan similar code to fix.